Sources/SPMTestSupport/Observability.swift

```diff
@@ -48,7 +48,7 @@ public struct TestingObservability {
         self.collector.hasWarnings
     }
 
-    struct Collector: ObservabilityHandlerProvider, DiagnosticsHandler, CustomStringConvertible {
+    final class Collector: ObservabilityHandlerProvider, DiagnosticsHandler, CustomStringConvertible {
```

Comment:
Making this a `class` because `diagnostics: ThreadSafeArrayStore` is a class instance, so there's no actual value semantics here even with a `struct`.